clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -660,12 +660,12 @@ private void processDisconnection(List<ClientResponse> responses,
                 AuthenticationException exception = disconnectState.exception();
                 connectionStates.authenticationFailed(nodeId, now, exception);
                 metadataUpdater.handleAuthenticationFailure(exception);
-                log.error("Connection to node {} failed authentication due to: {}", nodeId, exception.getMessage());
+                log.error("Connection to node {} ({}) failed authentication due to: {}", nodeId, disconnectState.remoteAddress(), exception.getMessage());
                 break;
             case AUTHENTICATE:
                 // This warning applies to older brokers which don't provide feedback on authentication failures
-                log.warn("Connection to node {} terminated during authentication. This may indicate " +
-                        "that authentication failed due to invalid credentials.", nodeId);
+                log.warn("Connection to node {} ({}) terminated during authentication. This may indicate " +
+                        "that authentication failed due to invalid credentials.", nodeId, disconnectState.remoteAddress());
                 break;
             case NOT_CONNECTED:
                 log.warn("Connection to node {} could not be established. Broker may not be available.", nodeId);
```

Comment:
Is it possible to include brokerIP/port in case of  normal connection failures? Currently we need to enable to debug logs to print the faulty ip/port.

Reply:
by "normal" you mean the NOT_CONNECTED case ?

Reply:
Yes, in case of java.net.ConnectException exceptions